app/views/events/show.html.erb

```diff
@@ -90,7 +90,14 @@
       <li class="list-group-item">There are no users registered for this event.</li>
     <% else %>
       <% @registered_users.each do |guest| %>
-        <li class="list-group-item" style="display: list-item;"><%= guest.name %> - <%= guest.email %></li>
+        <li class="list-group-item" style="display: list-item;">
+          <%= guest.name %> - <%= guest.email %>
+          <%= link_to 'Remove',
+            registration_path(user_id: guest.id, event_id: @event),
```

Comment:
Why would you expect params hash to contain `regitsration_id` when the parameters you are passing are:- user_id: guest.id, event_id: @event. anything written inside the parentheses are in params hash

Reply:
I see so `@registration = Registration.find(params[:registration_id])` is wrong it should be `@registration = Registration.find(params[:id])`. That's why I had to pass event_id and user_id

Reply:
<img width="860" alt="Screen Shot 2020-05-02 at 1 55 13 AM" src="https://user-images.githubusercontent.com/46115782/80856432-02a9af00-8c18-11ea-91e8-7d549197e100.png">

I know why `@user_registration` is `nil `for the admin page. It queries only for `current_user` registration. Basically, for user `params[:registration_id]` exists but not for admin. How do you want to optimize this?